src/plots/cartesian/layout_defaults.js

```diff
@@ -249,8 +249,9 @@ module.exports = function supplyLayoutDefaults(layoutIn, layoutOut, fullData) {
         handleTypeDefaults(axLayoutIn, axLayoutOut, coerce, defaultOptions);
         handleAxisDefaults(axLayoutIn, axLayoutOut, coerce, defaultOptions, layoutOut);
 
-        var unifiedHover = layoutIn.hovermode && ['x unified', 'y unified'].indexOf(layoutIn.hovermode) !== -1;
-        var unifiedSpike = unifiedHover && axLetter === layoutIn.hovermode.charAt(0);
+        var hovermode = layoutOut.hovermode || layoutIn.hovermode;
```

Comment:
Does it not work with just `layoutOut.hovermode`? Do we need the `layoutIn` fallback for some specific case?

Reply:
Some tests failed without the `layoutIn` fall back.

Reply:
If this is just unit tests (ie we're not running the entire `supplyDefaults` pipeline so earlier steps in that should have set `layoutOut.hovermode` were not called, then I'd rather change the test than include this fallback. If there's a case where the full pipeline does not behave correctly then this worries me, as that case presumably would not work with templates.

Reply:
`hovermode` coerce with auto defaults depending on `cartesian` is
https://github.com/plotly/plotly.js/blob/6de4c751a3d702ea13f20214349460840cfab6af/src/components/fx/layout_defaults.js#L36-L44
called at the end of pipeline here:
https://github.com/plotly/plotly.js/blob/6de4c751a3d702ea13f20214349460840cfab6af/src/plots/plots.js#L1672-L1678
I am not sure how to proceed if we want to remove `layoutIn.hovermode` fallback.

Reply:
On another note - here is test which is failing, if we remove `layoutIn` fallback:
https://github.com/plotly/plotly.js/blob/6de4c751a3d702ea13f20214349460840cfab6af/test/jasmine/tests/hover_label_test.js#L4037-L4052

Reply:
If that test is failing without the `layoutIn` fallback, it means that spikes will not be properly enabled by `hovermode` set in the `template`, right?

So I guess the code here is called by the `basePlotModules` loop:

https://github.com/plotly/plotly.js/blob/6de4c751a3d702ea13f20214349460840cfab6af/src/plots/plots.js#L1639-L1647

I think the easiest thing to do would be to move `hovermode` coercion here, right at the top of this function. That will involve making a special `coerce` function using `fx/layout_attributes`, but it will ensure the ordering is correct.

Reply:
@alexcjohnson Good call. Revised in the commits below.